app/controllers/conference_registrations_controller.rb

```diff
@@ -6,22 +6,19 @@ class ConferenceRegistrationsController < ApplicationController
 
   def new
     @registration = Registration.new(conference_id: @conference.id)
-    authorize! :new, @registration, message: "Sorry, you can not register for #{@conference.title}. Registration limit exceeded or the registration is not open."
 
     # Redirect to registration edit when user is already registered
     if @conference.user_registered?(current_user)
+      authorize! :edit, current_user.registrations.find_by(conference_id: @conference.id)
       redirect_to edit_conference_conference_registration_path(@conference.short_title)
       return
-    # ichain does not allow us to create users during registration
-    elsif (ENV['OSEM_ICHAIN_ENABLED'] == 'true') && !current_user
-      redirect_to root_path, alert: 'You need to sign in or sign up before continuing.'
-      return
     end
 
-    # avoid openid sign_in to redirect to register/new when the sign_in user had already a registration
-    if current_user && @conference.user_registered?(current_user)
-      redirect_to edit_conference_conference_registration_path(@conference.short_title)
+    unless @conference.registration_open? && !@conference.registration_limit_exceeded?
```

Comment:
An equivalent expression using can? that can replace the current one is the following: `(can? :new, @registration) || (can? :new, @registration) && ENV['OSEM_ICHAIN_ENABLED'] == 'true' && current_user == nil`
Also, code readability can be improved by replacing the unless statement with `if !@conference.registration_open? || @conference.registration_limit_exceeded?`, if that's an issue.